cis194/week3/robert-change/Golf/src/Golf.hs

```diff
@@ -14,5 +14,24 @@ import Data.List.HT
 --  but got: ["ABCD","AC","AD","A"]
 
 -- attempt 2 
+{-
 skips :: [a] -> [[a]]
-skips l = (flip $ (.) <$> sieve <*> drop . (-1 +)) l <$> [1..length l]
\ No newline at end of file
+skips l = (flip $ (.) <$> sieve <*> drop . (-1 +)) l <$> [1..length l]
+-}
+
+-- attempt 3 (inspired and copied from Zehua's solution)
+skips :: [a] -> [[a]]
+-- fmap: 
+-- applied n => sieve n (drop (n-1) l) to each element
+-- drop (n-1) l will remove the first n-1 elements
+-- then sieve n newElement will move the index to the current n
+-- so that attempt 1 issue will be resolved 
+skips l = (\n -> sieve n (drop (n-1) l)) <$> [1..length l]
+
+
+-- using recurison 
+localMaxima :: [Integer] -> [Integer]
+localMaxima (x:xs:y:z)
```

Comment:
nit:
In the production code base, we `xs` usually means `x`s. -- multiple x.
So here I suggests that 
```suggestion
localMaxima ( x : y : z : as )
```

Reply:
@shineli1984 what is the convention of the 4th parameter? `as` ?

Reply:
Since we got `x, y, z` I won't be fair to have a `xs` or `ys`. `zs` may be a good choice here but `as` is also acceptable.